### PR TITLE
Added models, cervices and repositories

### PR DESCRIPTION
**Pull request by maxedel**

### CODE REVIEW

File: flix-core/src/main/resources/db.migration/V__1.1__create_user_table.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE USER (
+  id        INTEGER,
+  login     VARCHAR(1000),
+  password  VARCHAR(50),
+  age       INTEGER,
+  gender    VARCHAR(10)
```
Comment (ManWithNoNamePL) [inline flix-core/src/main/resources/db.migration/V__1.1__create_user_table.sql]: PRIMARY KEY (USER_ID) - this should happen automatically, but better safe than sorry :)

File: flix-core/src/main/resources/db.migration/V__1.2__create_movie_table.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE MOVIE (
+  id        INTEGER,
+  title     VARCHAR(1000),
+  genre     VARCHAR(50),
+  runtime   INTEGER
```
Comment (ManWithNoNamePL) [inline flix-core/src/main/resources/db.migration/V__1.2__create_movie_table.sql]: ditto, PRIMARY KEY (MOVIE_ID)

File: flix-core/src/main/java/com/epam/mentoring/flixcore/model/User.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import javax.persistence.*;
+
+@Entity
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "ID")
+    protected Integer id;
```
[the remark this answers is not in the document]
Comment (maxedel) [reply flix-core/src/main/java/com/epam/mentoring/flixcore/model/User.java]: Done

File: flix-core/src/main/java/com/epam/mentoring/flixcore/model/User.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    private String login;
+
+    private String password;
+
+    private int age;
+
+    private String gender;
```
[the remark this answers is not in the document]
Comment (maxedel) [reply flix-core/src/main/java/com/epam/mentoring/flixcore/model/User.java]: Done

File: flix-core/src/main/resources/db.migration/V__1.1__create_user_table.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE USER (
+  id        INTEGER,
```
[the remark this answers is not in the document]
Comment (maxedel) [reply flix-core/src/main/resources/db.migration/V__1.1__create_user_table.sql]: Done

File: flix-core/src/main/resources/db.migration/V__1.2__create_movie_table.sql [hunk 6]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE MOVIE (
+  id        INTEGER,
```
[the remark this answers is not in the document]
Comment (maxedel) [reply flix-core/src/main/resources/db.migration/V__1.2__create_movie_table.sql]: Done

File: flix-core/src/main/resources/db.migration/V__1.2__create_movie_table.sql (hunk 2 above)
Comment (maxedel) [reply flix-core/src/main/resources/db.migration/V__1.2__create_movie_table.sql]: Done

File: flix-core/src/main/resources/db.migration/V__1.1__create_user_table.sql (hunk 1 above)
Comment (maxedel) [reply flix-core/src/main/resources/db.migration/V__1.1__create_user_table.sql]: Done

File: flix-core/src/main/java/com/epam/mentoring/flixcore/service/impl/MovieServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package com.epam.mentoring.flixcore.service.impl;
+
+import com.epam.mentoring.flixcore.model.Movie;
+import com.epam.mentoring.flixcore.model.User;
+import com.epam.mentoring.flixcore.repository.MovieRepository;
+import com.epam.mentoring.flixcore.service.MovieService;
+import org.springframework.beans.factory.annotation.Autowired;
+
```
[the remark this answers is not in the document]
Comment (maxedel) [reply flix-core/src/main/java/com/epam/mentoring/flixcore/service/impl/MovieServiceImpl.java]: Added

File: flix-core/src/main/java/com/epam/mentoring/flixcore/controller/MovieController.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    @Autowired
+    private MovieService movieService;
+
+    @RequestMapping(value = MOVIE_PATH_ROOT, method = RequestMethod.GET)
+    public ResponseEntity<List<Movie>> listAllMovies() {
+        List<Movie> movies = movieService.findAllMovies();
+        if(movies.isEmpty()){
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(movies, HttpStatus.OK);
```
Comment (ManWithNoNamePL) [inline flix-core/src/main/java/com/epam/mentoring/flixcore/controller/MovieController.java]: Again this is fine, but there is builder provided that may be easily used.
For this sample something like:
return ResponseEntity.ok(movies).build();

File: flix-core/src/main/java/com/epam/mentoring/flixcore/controller/MovieController.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.List;
+
+@Controller
```
[the remark this answers is not in the document]
Comment (maxedel) [reply flix-core/src/main/java/com/epam/mentoring/flixcore/controller/MovieController.java]: Done

File: flix-core/src/main/java/com/epam/mentoring/flixcore/controller/MovieController.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+public class MovieController {
+
+    private static final String MOVIE_PATH_ROOT = "/movie/";
+
+    @Autowired
+    private MovieService movieService;
+
+    @RequestMapping(value = MOVIE_PATH_ROOT, method = RequestMethod.GET)
```
[the remark this answers is not in the document]
Comment (maxedel) [reply flix-core/src/main/java/com/epam/mentoring/flixcore/controller/MovieController.java]: Done

File: flix-core/src/main/java/com/epam/mentoring/flixcore/controller/MovieController.java (hunk 8 above)
Comment (maxedel) [reply flix-core/src/main/java/com/epam/mentoring/flixcore/controller/MovieController.java]: Done

**Pull request merged**